crates/apub/src/api/user_settings_backup.rs

```diff
@@ -0,0 +1,382 @@
+/// To import a larger backup, split it into multiple parts.
+///
+/// TODO: having the user manually split files will very be confusing
+const MAX_URL_IMPORT_COUNT: usize = 200;
+
+/// Backup of user data. This struct should never be changed so that the data can be used as a
+/// long-term backup in case the instance goes down unexpectedly. All fields are optional to allow
+/// importing partial backups.
+///
+/// This data should not be parsed by apps/clients, but directly downloaded as a file.
+///
+/// Be careful with any changes to this struct, to avoid breaking changes which could prevent
+/// importing older backups.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct UserBackup {
+  pub display_name: Option<String>,
+  pub bio: Option<String>,
+  pub avatar: Option<DbUrl>,
+  pub banner: Option<DbUrl>,
+  pub matrix_id: Option<String>,
+  pub bot_account: Option<bool>,
+  // TODO: might be worth making a separate struct for settings backup, to avoid breakage in case
+  //       fields are renamed, and to avoid storing unnecessary fields like person_id or email
+  pub settings: Option<LocalUser>,
+  #[serde(default)]
+  pub followed_communities: Vec<ObjectId<ApubCommunity>>,
+  #[serde(default)]
+  pub blocked_communities: Vec<ObjectId<ApubCommunity>>,
+  #[serde(default)]
+  pub blocked_users: Vec<ObjectId<ApubPerson>>,
+  #[serde(default)]
+  pub blocked_instances: Vec<String>,
+}
+
+#[tracing::instrument(skip(context))]
+pub async fn export_user_backup(
+  local_user_view: LocalUserView,
+  context: Data<LemmyContext>,
+) -> Result<Json<UserBackup>, LemmyError> {
+  let lists = LocalUser::export_backup(&mut context.pool(), local_user_view.person.id).await?;
+
+  Ok(Json(UserBackup {
+    display_name: local_user_view.person.display_name,
+    bio: local_user_view.person.bio,
+    avatar: local_user_view.person.avatar,
+    banner: local_user_view.person.banner,
+    matrix_id: local_user_view.person.matrix_user_id,
+    bot_account: local_user_view.person.bot_account.into(),
+    settings: Some(local_user_view.local_user),
+    followed_communities: lists
+      .followed_communities
+      .into_iter()
+      .map(Into::into)
+      .collect(),
+    blocked_communities: lists
+      .blocked_communities
+      .into_iter()
+      .map(Into::into)
+      .collect(),
+    blocked_users: lists.blocked_users.into_iter().map(Into::into).collect(),
+    blocked_instances: lists.blocked_instances,
+  }))
+}
+
+#[tracing::instrument(skip(context))]
+pub async fn import_user_backup(
+  data: Json<UserBackup>,
+  local_user_view: LocalUserView,
+  context: Data<LemmyContext>,
+) -> Result<Json<SuccessResponse>, LemmyError> {
+  let display_name = Some(sanitize_html_api_opt(&data.display_name));
+  let bio = Some(sanitize_html_api_opt(&data.bio));
+
+  let person_form = PersonUpdateForm {
+    display_name,
+    bio,
+    // TODO: might want to reupload avatar and banner to local instance
```

Comment:
I'd say just null / ignore them. Let image uploads be explicit, let this be a text data import only.